### PR TITLE
[benchmark] add cpuprofiler flag to save a profiling session

### PR DESCRIPTION
**Pull request by palazzem**
### What it does

* add `-cpuprofile=cpu.pprof` flag
* add `-memprofile=mem.pprof` flag

It's useful to profile the full flow overhead when the agent is constantly hit by a huge number of traces per second.

### CODE REVIEW

File: agent/main.go [hunk 1]
```diff
@@ -72,13 +75,29 @@ func versionString() string {
...
+	// command-line arguments
 	flag.StringVar(&opts.ddConfigFile, "ddconfig", "/etc/dd-agent/datadog.conf", "Classic agent config file location")
 	// FIXME: merge all APM configuration into dd-agent/datadog.conf and deprecate the below flag
 	flag.StringVar(&opts.configFile, "config", "/etc/datadog/trace-agent.ini", "Trace agent ini config file.")
 	flag.BoolVar(&opts.debug, "debug", false, "Turn on debug mode")
 	flag.BoolVar(&opts.version, "version", false, "Show version information and exit")
+
+	// profiling arguments
+	cpuprofile := flag.String("cpuprofile", "", "write cpu profile to file")
+	memprofile := flag.String("memprofile", "", "write memory profile to `file`")
```
Comment (LeoCavaille) [inline agent/main.go]: Not sure why these are handled differently, they are command-line arguments too?
Also I wonder if we should go this way or prefer signaling to profile a live process. Like you send SIGUSR1 to start and stop a profile (we have that for some of our python stuff).

File: agent/main.go [hunk 2]
```diff
@@ -5,15 +5,18 @@ import (
...
+	"runtime"
+	"runtime/pprof"
 	"syscall"
 	"time"
 
 	"github.com/DataDog/datadog-trace-agent/config"
 	"github.com/DataDog/datadog-trace-agent/statsd"
 	log "github.com/cihub/seelog"
+
+	_ "net/http/pprof"
```
Comment (LeoCavaille) [inline agent/main.go]: Do we need this? I think there is already a pprof server running in the receiver at localhost:7777/debug/pprof/ ?

File: agent/main.go (hunk 1 above)
Comment (LeoCavaille) [inline agent/main.go]: Actually scratch that, we can already get that via pprof. Can't we do long running profiles with it already?

File: agent/main.go (hunk 2 above)
Comment (palazzem) [inline agent/main.go]: yes that one can be removed

File: agent/main.go (hunk 1 above)
Comment (palazzem) [inline agent/main.go]: I've used these flags only to have a single entrypoint to:
* start the agent with ``-cpuprofile``
* start the ``trace-flood`` command that hits the agent a huge number of times
* wait for some seconds (profiling is activated)
* stop everything
* read the profile file

I can remove everything, but in this way (I think) it's easy to script and to run locally. To profile a running execution we can just use the pprof HTTP.

File: agent/main.go (hunk 2 above)
Comment (palazzem) [inline agent/main.go]: Wait.... we don't have a ``_ "net/http/pprof"`` in the receiver. If I remove that line:
```
$ curl localhost:7777/debug/pprof/
404 page not found
```

File: agent/main.go (hunk 1 above)
Comment (palazzem) [inline agent/main.go]: I'm currently using ``forego`` with the following ``Procfile``:
```
agent: ./trace-agent -cpuprofile cpu.prof
flood: ./trace-flood
interrupt: sleep 60; exit 1
```

In this way I could run the profiler for X seconds and see profiling results.

**Comment by palazzem** [comment]
This PR has been merged with #160 in a new branch: https://github.com/DataDog/datadog-trace-agent/compare/master...palazzem-ufoot/profiling-branch

Closing since it will not land on ``master`` for the moment.

**Pull request closed without merging**